vue-thacer/src/assets/js/thacer-map-create-layer.js

```diff
@@ -64,71 +64,80 @@ export function setCeramLayer(ceramLayer) {
-      // function select ceram according to the secteurs ID on the clicked secteur feature
-      e.target.eachLayer(function (layer) {
-        layer.eachLayer(function (e) {
-          let string = ''
-          let biblio = ''
-          if (
-            !(
-              // GTh means "Guide de Thasos"
-              (
-                (layer.feature.properties.GTh == '') |
-                (layer.feature.properties.GTh == 'null') |
-                (layer.feature.properties.GTh == undefined)
-              )
-            )
-          ) {
-            string = ' GTh' + layer.feature.properties.GTh
-          }
-          if (
-            !(
-              (layer.feature.properties.Référenc == '') |
-              (layer.feature.properties.Référenc == 'null') |
-              (layer.feature.properties.Référenc == undefined)
+  // create a new leaflet GeoJSON layer
+  const layer = L.geoJSON();
+
+  // fetch the data from the API URL using the fetch method
+  fetch(import.meta.env.VITE_API_URL + 'geojson/secteurs.geojson')
+    .then(response => response.json()) // parse the response as JSON
+    .then(data => {
+      // add the GeoJSON data to the leaflet layer
+      layer.addData(data);
+
+      // set up the popup and click events for each layer
+      layer.eachLayer(function (e) {
+        let string = ''
+        let biblio = ''
+        if (
+          !(
+            // GTh means "Guide de Thasos"
+            (
+              (e.feature.properties.GTh == '') |
+              (e.feature.properties.GTh == 'null') |
+              (e.feature.properties.GTh == undefined)
             )
-          ) {
-            biblio = layer.feature.properties.Référenc
-          }
-          e.bindPopup(layer.feature.properties.Titre + string + '<br>' + biblio, {
-            maxWidth: 300,
-            minWidth: 10,
-            maxHeight: 250,
-            autoPan: true,
-            closeButton: false,
-            autoPanPadding: [0, 0]
-          })
-          // search ceram on click
-          e.on('click', function () {
-            search.searchCeramByClick(ceram, markerClusterGroupCeram, map, layer)
-          })
+          )
+        ) {
+          string = ' GTh' + e.feature.properties.GTh
+        }
+        if (
+          !(
+            (e.feature.properties.Référenc == '') |
+            (e.feature.properties.Référenc == 'null') |
+            (e.feature.properties.Référenc == undefined)
+          )
+        ) {
+          biblio = e.feature.properties.Référenc
+        }
+        e.bindPopup(e.feature.properties.Titre + string + '<br>' + biblio, {
+          maxWidth: 300,
+          minWidth: 10,
+          maxHeight: 250,
+          autoPan: true,
+          closeButton: false,
+          autoPanPadding: [0, 0]
         })
-      })
-    })
+        // search ceram on click
+        e.on('click', function () {
+          search.searchCeramByClick(ceram, markerClusterGroupCeram, map, e)
+        })
+      });
+    });
```

Comment:
@chtzv ah, your config is definitely not using prettier, where it is asking to remove semicolon. Please check the readme, and don't hesitate to ask me. 
You can run a `npm run format`

Reply:
@francoiscrouzet , I followed the readme, and I have volar, prettier - code formatter, and eslint installed and enabled. Pretrier is the default formatter and it does format on save. Prettier works but it does not enforce removal of semicolons. I tried disabling all extensions but prettier. The result is the same.  
However, if I run Format document (forced) the semicolons are gone.
Below are the reports from running 1) "Format" and 2) "Format (forced)". They are the same. I do not have an ignore file. It's a mystery to me... 

["INFO" - 11:58:13 PM] Extension Name: esbenp.prettier-vscode.
["INFO" - 11:58:13 PM] Extension Version: 9.10.4.
["INFO" - 12:05:11 AM] Forced formatting will not use ignore files.
["INFO" - 12:05:11 AM] Formatting file:///home/chavdar/Thacer/thacer/vue-thacer/src/assets/js/thacer-map-create-layer.js
["INFO" - 12:05:12 AM] Using config file at '/home/chavdar/Thacer/thacer/vue-thacer/.prettierrc.json'
["INFO" - 12:05:12 AM] Using ignore file (if present) at /home/chavdar/Thacer/thacer/.prettierignore
["INFO" - 12:05:12 AM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 12:05:12 AM] Detected local configuration (i.e. .prettierrc or .editorconfig), VS Code configuration will not be used
["INFO" - 12:05:12 AM] Prettier Options:
{
  "filepath": "/home/chavdar/Thacer/thacer/vue-thacer/src/assets/js/thacer-map-create-layer.js",
  "parser": "babel",
  "semi": false,
  "tabWidth": 2,
  "singleQuote": true,
  "printWidth": 100,
  "trailingComma": "none"
}
["INFO" - 12:05:12 AM] Formatting completed in 527ms.


["INFO" - 12:35:25 AM] Forced formatting will not use ignore files.
["INFO" - 12:35:25 AM] Formatting file:///home/chavdar/Thacer/thacer/vue-thacer/src/assets/js/thacer-map-create-layer.js
["INFO" - 12:35:25 AM] Using config file at '/home/chavdar/Thacer/thacer/vue-thacer/.prettierrc.json'
["INFO" - 12:35:25 AM] Using ignore file (if present) at /home/chavdar/Thacer/thacer/.prettierignore
["INFO" - 12:35:25 AM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 12:35:25 AM] Detected local configuration (i.e. .prettierrc or .editorconfig), VS Code configuration will not be used
["INFO" - 12:35:25 AM] Prettier Options:
{
  "filepath": "/home/chavdar/Thacer/thacer/vue-thacer/src/assets/js/thacer-map-create-layer.js",
  "parser": "babel",
  "semi": false,
  "tabWidth": 2,
  "singleQuote": true,
  "printWidth": 100,
  "trailingComma": "none"
}
["INFO" - 12:35:25 AM] Formatting completed in 252ms.

Reply:
Hi Chavdar, it seems your vsc doesn't find the .prretierrc.json file, and use the default config. (You may see that by clicking in the bottom right of vscode, on the `prettier` label. It opens a terminal with some info about prettier. see 1st screenshot)
You need to specify it, and it should work, see 2nd screenshot.
If it solves your problem, would you mind adding this info to the readme in this PR ?   
   
<img width="1352" alt="Screenshot 2023-04-19 at 08 29 21" src="https://user-images.githubusercontent.com/2144869/232985790-87a2991b-0c2e-4b4e-b0f5-062327194cbf.png">

    
<img width="827" alt="Screenshot 2023-04-19 at 08 24 45" src="https://user-images.githubusercontent.com/2144869/232985531-a72c3db1-3cd3-4223-b19d-9ea13999b9b7.png">

Reply:
@francoiscrouzet, the info I pasted above is exactly from Prettier terminal. It shows that Prettier finds the configuration file. Adding the path to the settings does not change anything. It is something else. I think that for some reason Prettier ignores the files that it is supposed to format.

Reply:
@francoiscrouzet , I solved it. Misconfiguration in the VSC configuration file, indicating a different formatter specifically for JavaScript.

Reply:
Great. In case you think the readme was not cleat enough, or that this info could help other people joining the project, don't hesitate to update the reamde